### ISSUE TITLE
Publication State isn't respected in Relational Fields when making a graphql query

### ISSUE THREAD

**Issue by galago** [strapi/strapi#15682]
Required System information

Node.js version: 16
NPM version: 6
Strapi version: 4.6.0
Database: SQLite
Operating system: Windows 10

### Describe the bug
Publication State isn't applied in relational fields when making a graphql query, and furthermore, there's no explicit way to do it even forcing `publicationState: LIVE` in the query. Draft results will always show no matter what. 

Making a GraphQL query, querying, and noticing that the publication state of the related elements on the component is not respected.

See the screenshot how even with the publicationState argument, the nonpublished element is retrieved ("publishedAt": null)

![Captura de pantalla 2023-02-02 162108](https://user-images.githubusercontent.com/1188316/216283441-e90839f6-df32-4b08-92a3-0d49ab26acf3.png)

![Captura de pantalla 2023-02-02 161850](https://user-images.githubusercontent.com/1188316/216283183-47a468ca-2e51-4ad6-bcc1-bb0da567c458.png)

Expected behavior
For all element publication state to be respected.

**Comment by Marc-Roig** [comment]
Was able to reproduce this, indeed it seems publicationState is not applied to relational fields.

A workaround could be done using this filter instead:
```
    "publishedAt": {
      "notNull": true
    }
```

**Comment by joshuaellis** [comment]
@Marc-Roig do you think you could handle this? If not we'll leave it to DX.

**Comment by Marc-Roig** [comment]
I can handle this.

The problem is in the entity-service,  the [load function](https://github.com/strapi/strapi/blob/253ba050a7322534457ead6cc00a3596f3d690fa/packages/core/strapi/lib/services/entity-service/index.js#L279) does not transform the publication state to the correct filter

**Comment by Marc-Roig** [comment]
Opened a PR with a solution for this :) , we will try to merge this for 4.6.1. Thank you @galago  for the issue description, helped me identify the issue very fast.

**Comment by galago** [comment]
Thanks so much for the prompt response. Hope to see it fixed in the next version, keep up the great work!

**Comment by Marc-Roig** [comment]
This should be fixed for the next 4.6.2 release this week 😄

**Comment by DavidForDev** [comment]
@Marc-Roig 

**Hey I have a similar issue:** 

I'm facing an issue with content creation and publishing in Strapi. When I create a new wine product and publish it, two entries appear in the database:

Draft Entry: publishedAt is null.
Published Entry: publishedAt has a valid timestamp.
Workflow:
Creating a New Wine: I create a product in the Strapi admin panel.
Publishing the Wine: I publish the product after saving.
Checking the database (data.db), I find both entries, leading to confusion.

**Example GraphQL Query:**

```
query Query($status: PublicationStatus) {
  wines(status: $status) {
    documentId
    title
    createdAt
    updatedAt
    publishedAt
  }
}
```

**Variables:**

```
For drafts: { "status": "DRAFT" }
For published: { "status": "PUBLISHED" }

```


**Observed Behavior:**

Querying drafts returns both draft and published entries.
Multiple entries for the same wine product complicate data management.

**What I did to get the meaning of the problem:**

when I checked ` .tmp=>db` file I saw that after publishing "wine" it duplicates data where one of wine's publishedAt is null, I think it should just update wine and not duplicate.

**Questions:**

1) Is this expected behavior in Strapi?
2) How can I retrieve only published entries without duplicates?
3) What are the best practices for managing content lifecycle to avoid duplicates?

**Thank you for your help!**
